app/services/import_service.rb

```diff
@@ -124,6 +126,28 @@ def import_bookmarks!
     end
   end
 
+  def import_lists!
+    parse_import_data!(['List name', 'Account address'])
+    items = @data.take(ROWS_PROCESSING_LIMIT).map { |row| [row['List name'], row['Account address']] }
+    list_titles = items.map { |row| row[0] }.uniq
+
+    if @import.overwrite?
+      List.where(title: list_titles, account: @account).destroy_all
+    end
+
+    items.each do |listname, accountname|
+      list = @account.lists.find_or_create_by!(account: @account, title: listname)
```

Comment:
More queries than necessary in this implementation. I would first convert the flat list to a list of accounts grouped by list title, then iterate over the list titles -- that way you only find-or-create each list once.